tests/learner/learner_test.py

```diff
@@ -39,30 +36,57 @@ def test_subset_learner_ball():
             ball_template, chooser=PHASE1_CHOOSER, ontology=GAILA_PHASE_1_ONTOLOGY
         ),
     )
-    colored_truck_object = object_variable(
-        "truck-with-color", TRUCK, added_properties=[color_variable("color")]
+
+    learner = SubsetLanguageLearner()
+    for training_stage in [ball_curriculum]:
+        for (
+            _,
+            linguistic_description,
+            perceptual_representation,
+        ) in training_stage.instances():
+            learner.observe(
+                LearningExample(perceptual_representation, linguistic_description)
+            )
+
+    for test_instance_group in [test_ball_curriculum]:
+        for (
+            _,
+            test_instance_language,
+            test_instance_perception,
+        ) in test_instance_group.instances():
+            descriptions_from_learner = learner.describe(test_instance_perception)
+            gold = test_instance_language.as_token_sequence()
+            assert [desc.as_token_sequence() for desc in descriptions_from_learner][
+                0
+            ] == gold
+
+
+def test_subset_learner_dog_infinite_loop():
```

Comment:
Does `dog` still produce an infinite loop?  (or take excessively long?) If not we should remove this test.